store/sqlstore/link_metadata_store.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package sqlstore
+
+import (
+	"database/sql"
+	"net/http"
+
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/mattermost/mattermost-server/store"
+)
+
+type SqlLinkMetadataStore struct {
+	SqlStore
+}
+
+func NewSqlLinkMetadataStore(sqlStore SqlStore) store.LinkMetadataStore {
+	s := &SqlLinkMetadataStore{sqlStore}
+
+	for _, db := range sqlStore.GetAllConns() {
+		table := db.AddTableWithName(model.LinkMetadata{}, "LinkMetadata").SetKeys(false, "Hash")
+		table.ColMap("URL").SetMaxSize(512)
+		table.ColMap("Type").SetMaxSize(16)
+		table.ColMap("Data").SetMaxSize(4096)
```

Comment:
And likewise where does 4096 come from? Why not use text type?

Reply:
It's arbitrary, based off what we use for fields like the post Props. The MySQL driver actually forces this to be Text, but it seems the Postgres one just does a varchar(4096). I can change it if we have any strong motivation to, but I'd rather just match how we do it elsewhere for now

Reply:
Under the hood, in postgres, a varchar is a text with a limitation in the stored data. I prefer to have a limit there because if not, you can generate a opengraph response from your webpage with a size of 3GB and then, you pase 100 links with a `url?id=random-number` and boom, DOS. Actually, I would limit the max bytes read during the opengraph requests to avoid this.

Reply:
I know we do this elsewhere, but the problem with doing this at the database tier using a `varchar(N)` vs using `text` with an app-tier limit of length is that if we find that the size isn't sufficient we're forced to do a more difficult DB migration vs simply changing a constant. And since it doesn't make a difference in terms of query efficiency or storage used I don't see why we keep keep introducing this idiom.

Reply:
I agree that it might make sense to change this across the board, but it would be better to discuss this when we have more time. As it is, this PR will hopefully be what's required for us to get post metadata out of experimental which would unblock the rest of the scrolling improvements.

Maybe file a dev meeting item to discuss this? At least for me, a lot of our database decisions feel arbitrary, so it might be good to give things like this some more thought